mysql-test/main/information_schema_temp_table.test

```diff
@@ -278,3 +278,36 @@ drop table seq2;
 show  full tables;
 drop table seq3;
 show  full tables;
+
+--echo #
+--echo # MDEV-32973: SHOW TABLES LIKE shows temporary tables with
+--echo #             non-matching names
+--echo #
+
+CREATE DATABASE db;
+USE db;
+CREATE TABLE realtable (a INT);
+CREATE TEMPORARY TABLE temptable (a INT);
+SHOW TABLES LIKE 'faketable';
+SHOW TABLES LIKE 'temptable';
+SHOW TABLES LIKE '%pt%';
+SHOW TABLES LIKE '%lt%';
+select table_name from information_schema.tables where
+  table_schema='db' and table_name='temptable';
+select table_name from information_schema.tables where
+  table_schema='db' and table_name='faketable';
+CREATE TEMPORARY TABLE t2 (a INT);
+select table_name from information_schema.tables where
+  table_schema='db' and table_name='T2';
+SHOW TABLES LIKE 'T%';
+create  table tmp (a int);
+create  table t_p (a int);
```

Comment:
this should've been _create **temporary** table_.
but anyway, it won't show the problem, because the server apparently rechecks the condition later and even if your code allows unrelated tables in, the server will filter them out later.

So, remove this test case completely (with `tmp` and `t_p`) and try this:
```sql
create temporary table `t\_p` (a int);
select table_name from information_schema.tables where
  table_schema='db' and table_name='t\\_p';
drop table `t\_p`;
```

Try with both temporary and non-temporary table to see the difference in behavior

Reply:
Hi @vuvova,
I debugged and I don't see way how to:
1. catch difference for temporary tables between 
- a)
```sql
create temporary table `t\_p` (a int); select table_name from information_schema.tables where table_schema='db' and table_name='t\\_p'; 
# Result should be `t\_p`
```
- b)
```sql
CREATE TEMPORARY TABLE t2 (a INT);
select table_name from information_schema.tables where
  table_schema='db' and table_name='T2';
```
Both queries are currently failing in `wild_compare()` and both are respecting following conditions:
```
  if (!plan->lookup_field_vals.wild_table_value && schema_table_idx == SCH_TABLES)
```

2. compare table names for the regular tables.
    Regular tables are using `make_db_list()` and `make_table_name_list()`, but  I don't see when and where is `wild_compare()` checked (have debugged couldn't find stack trace for this debug point, but we know it works).
 
Can you give some hint as  direction to check please?

Reply:
sure.
```sql
create database `t\_p`;
create temporary table `t\_p`.tmp (a int);
select table_schema, table_name from information_schema.tables where
  table_schema='t\\_p';
```
_Disclaimer: untested_